File: backend/services/core-service/src/main/java/kr/lastdish/core/dish/domain/DishStatus.java
package kr.lastdish.core.dish.domain;

public enum DishStatus {
  ON_SALE("판매중"),
  SOLD_OUT("품절"),
  CLOSED("마감"),
  EXPIRED("만료");

  private final String displayName;

  DishStatus(String displayName) {
    this.displayName = displayName;
  }

  public String getDisplayName() {
    return displayName;
  }
}

File: frontend/lib/domain/model/dish_status.dart
/// 상품 판매상태(백엔드 `DishStatus` enum). order_status.dart/category.dart와 같은 패턴.
const List<String> dishStatusValues = ['ON_SALE', 'SOLD_OUT', 'CLOSED', 'EXPIRED'];

const Map<String, String> dishStatusDisplayNames = {
  'ON_SALE': '판매중',
  'SOLD_OUT': '품절',
  'CLOSED': '마감',
  'EXPIRED': '만료',
};

String dishStatusLabel(String value) => dishStatusDisplayNames[value] ?? value;
